content/2022/prepoznavanje-znakovnog-jezika.md

```diff
@@ -2,3 +2,105 @@
+
+>>>>>>> Stashed changes
+![ASL](/images/2022/prepoznavanje-znakovnog-jezika/acab.PNG)
+
+#### Obrada baze za kNN
+
+Za detektovanje celog regiona slike na kome se nalazi šaka, a samim tim i klasifikovanje znaka šake, tražena je boja kože. Vrednost ove boje će biti predstavljena kao opseg - različit je za svaku sliku jer se u bazi mogu pronaći slike sa senkama i slike šaka drugačijih tonova kože.
+Prvi pristup za utvrđivanje boje kože je uzimanje srednje vrednosti dobijenih 42 tačaka, pri čemu su dobijeni neprecizni rezultati. Drugi način bio je odredjivanje koordinate sredine šake i uzimanje njene vrednosti, što nije radilo jer se često nalazila senka na tom delu slike. Finalni i najprecizniji način je bio uzimanje celog opsega ovih tačaka.
+Na osnovu HSV vrednosti u koordinatama ključnih tačaka određen je spektar za koji klasifikujemo tačku kao da pripada šaci.
+<<<<<<< Updated upstream
+=======
+
+Za identifikaciju regiona slike koji je boje kože, prvo je korišćen MediaPipe Holistic pipeline za ekstraktovanje 21 ključnih tačaka obe šake (ukupno 42 koordinata). Na osnovu HSV vrednosti u koordinatama tačaka određen je spektar za koji klasifikujemo tačku kao da pripada šaci.
+>>>>>>> Stashed changes
+
+$$[HSVmin, HSVmax] = [min(kp_{1},kp_{2},...,kp_{n}), max(kp_{1},kp_{2},...,kp_{n})]$$
+
+Na osnovu dobijene pozicije šake slika je isečena oko nje i preoblikovana na 512 x 512 piksela. Na novodobijenu sliku primenjena je binarizacija i morfološke operacije radi uklanjanja šuma.
+
+#### Obrada baze za ASL Keypoint Classification
+
+Baza ove mreže se sastoji od koordinata ključnih tačaka šake. Za njihovu detekciju korišćena je MediaPipe Holistic metoda. Prilikom njene primene treshold za detektovanje tačaka šake je smanjen jer za određene položaje šake ona nije bivala detektovana. Na svaku sliku baze je pojedinačno primenjena detekcija ključnih tačaka i novodobijene slike su sačuvane u novu bazu. Nova baza je imala 63 parametra za svaku sliku:
+
+- X - x koordinata svake tačke, 21 ukupno
+- Y - y koordinata svake tačke, 21 ukupno
+- Z - dubina svake tačke, 21 ukupno
+
+### kNN - k Nearest Neighboors
+
+Za detektovanje celog regiona slike na kome se nalazi šaka, a samim tim i klasifikovanje znaka šake, tražena je boja kože. Vrednost ove boje će biti predstavljena kao opseg - različit je za svaku sliku jer se u bazi mogu pronaći slike sa senkama i slike šaka drugačijih tonova kože.
+
+Binarizovana slika je podeljena na NxN grid, gde definišemo obeležje svakog bloka kao zasićenost - odnos broja belih piskela i cele slike:
+
+$$obeležje = \frac{P_{bela}}{P_{blok}}$$
+
+Pošto su ovako dobijeni podaci podeljeni na klastere, potreban je efikasan način za njihovu klasifikaciju. Za ovo je korišćen KNN algoritam. Na prethodno fitovanim podacima u Euklidskoj metrici je izdračunata distanca između uzorka i svih fitovanih uzoraka i izabrani su k najbližih. Od tih k uzoraka klasifikator bira klasu koja se najčešće pojavljuje.
```

Comment:
Ovu rečenicu totalno promenite, ne možete da tvrdite da je KNN efikasan način za klasifikaciju. Pomenite da koristite euklidsku udaljenost. Koliko imate primeraka svake klase u setu koji koristitte za klasifikaciju znaka